internal/content_coding.go

```diff
@@ -166,7 +168,9 @@ func (d *GzipDecoder) Decode(data []byte) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	return d.buf.Bytes(), nil
+	out := make([]byte, d.buf.Len())
+	copy(out, d.buf.Bytes())
```

Comment:
TBH I don't like this to be in the encoder/decoder as it now adds a copy operation for everyone even though we might not use the encoder/decoder in parallel... Can you please move the copy operation to your plugin's code!

Reply:
Let me take this to the core-devs to discuss this. Will report back here...

Reply:
Moved the copy operation to pubsub plugin's code https://github.com/influxdata/telegraf/pull/9159/commits/809f650f099438987792b495e0c8c3bd4426130d